python/grapher.go

```diff
@@ -34,33 +36,64 @@ func NewGraphContext(unit *unit.SourceUnit) *GraphContext {
 // Graphs the Python source unit. If run outside of a Docker container, this assumes that the source unit has already
 // been installed (via pip or `python setup.py install`).
 func (c *GraphContext) Graph() (*graph.Output, error) {
-	if os.Getenv("IN_DOCKER_CONTAINER") != "" {
-		// NOTE: this may cause an error when graphing any source unit that depends
-		// on jedi (or any other dependency of the graph code)
-		requirementFiles, err := filepath.Glob(filepath.Join(c.Unit.Dir, "*requirements*.txt"))
+	programMode := os.Getenv("IN_DOCKER_CONTAINER") == ""
+	tc, err := toolchain.Lookup("sourcegraph.com/sourcegraph/srclib-python")
+	if err != nil {
+		return nil, err
+	}
+	pipBin := "pip"
+	pythonBin := "python"
+
+	if programMode {
+		tempPath, err := getTempPath()
 		if err != nil {
 			return nil, err
 		}
-		runCmdLogError(exec.Command("pip", "install", "-I", c.Unit.Dir))
-		for _, requirementFile := range requirementFiles {
-			err := runCmdStderr(exec.Command("pip", "install", "-r", requirementFile))
-			if err != nil {
-				log.Printf("Error installing dependencies in %s. Trying piecemeal install")
-				if b, err := ioutil.ReadFile(requirementFile); err == nil {
-					for _, req := range strings.Split(string(b), "\n") {
-						runCmdLogError(exec.Command("pip", "install", req))
-					}
-				} else {
-					log.Printf("Could not read %s: %s", requirementFile, err)
-				}
+		envName := fmt.Sprintf("%s-%s-env", getHash(c.Unit.Dir), url.QueryEscape(c.Unit.Name))
+		envDir := filepath.Join(tempPath, envName)
```

Comment:
`make .env` was never really intended to be called manually. You should do `make install` like before, that will install all the dependencies including virtualenv.

`make .env` creates empty virtualenv.